src/Jss.js

```diff
@@ -74,22 +75,33 @@ export default class Jss {
    * @see createRule
    * @api public
    */
-  createRule(selector, style, options) {
-    // Enable rule without selector.
-    if (typeof selector == 'object') {
+  createRule(name, style, options) {
+    // Enable rule without name for inline styles.
+    if (typeof name === 'object') {
       options = style
-      style = selector
-      selector = null
+      style = name
+      name = null
     }
-    const rule = createRule(selector, style, {
+
+    options = {
       jss: this,
       Renderer: findRenderer(options),
       ...options
-    })
-    this.plugins.run(rule)
+    }
+
+    const rule = this.factory.get(name, style, options)
+
+    if (options.applyPlugins !== false) {
+      this.plugins.handleRule(rule)
+    }
+
     return rule
   }
 
+  registerRuleClass(name, cls) {
```

Comment:
I think that this method (`registerRuleClass`) and the injectable `generateClassName` function are unicorns that feel a little bolted on and out of place.